Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -70,42 +71,72 @@ async function main() {
             var webPackage = packageUtility.PackageUtility.getPackagePath(taskParams.Package);
             var isFolderBasedDeployment = deployUtility.isInputPkgIsFolder(webPackage);
             var physicalPath: string = '/site/wwwroot';
-            if(taskParams.VirtualApplication) {
+            if (taskParams.VirtualApplication) {
                 physicalPath = await appServiceUtility.getPhysicalPath(taskParams.VirtualApplication);
                 await kuduServiceUtility.createPathIfRequired(physicalPath);
                 virtualApplicationPath = physicalPath;
             }
 
             webPackage = await FileTransformsUtility.applyTransformations(webPackage, taskParams);
 
-            if(deployUtility.canUseWebDeploy(taskParams.UseWebDeploy)) {
+            if (deployUtility.canUseWebDeploy(taskParams.UseWebDeploy)) {
                 tl.debug("Performing the deployment of webapp.");
-                if(!tl.osType().match(/^Win/)){
+                if (!tl.osType().match(/^Win/)) {
                     throw Error(tl.loc("PublishusingwebdeployoptionsaresupportedonlywhenusingWindowsagent"));
                 }
 
-                if(taskParams.RenameFilesFlag) {
+                if (taskParams.RenameFilesFlag) {
                     await appServiceUtility.enableRenameLockedFiles();
                 }
 
                 var msDeployPublishingProfile = await appServiceUtility.getWebDeployPublishingProfile();
-                await msDeploy.DeployUsingMSDeploy(webPackage, taskParams.WebAppName, msDeployPublishingProfile, taskParams.RemoveAdditionalFilesFlag,
-                    taskParams.ExcludeFilesFromAppDataFlag, taskParams.TakeAppOfflineFlag, taskParams.VirtualApplication, taskParams.SetParametersFile,
-                    taskParams.AdditionalArguments, isFolderBasedDeployment, taskParams.UseWebDeploy);
+                // this is being done temporarily, to tackle war files not getting expanded (only for the first deployment to the web app) when done using MSDeploy.
+                if (webPackage.toString().toLowerCase().endsWith('.war')) {
```

Comment:
Please add a separate file under operationsFolder. (say MSdeployOperations.ts) and place your logic there.